spec/datadog/appsec/component_spec.rb

```diff
@@ -0,0 +1,49 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/component'
+
+RSpec.describe Datadog::AppSec::Component do
+  describe '.build_appsec_component' do
+    context 'when appsec is enabled' do
+      it 'returns a Datadog::AppSec::Component instance' do
+        settings = double('AppSec::Settings', appsec: double('enabled', enabled: true))
```

Comment:
I agree with that. I initialled started using `AppSec::Configuration::Settings`. The problem is there is no way to override the values of the settings unless using ENV variables, which would mean I would have to mock `ENV`, which is fine, but I thought just not doing that and going the easy path. 

I will add a note to myself to improve `AppSec::Configuration::Settings` to allow to accept some kind of overrides so it would improve testing in the future. 

In the meantime, I will change the test to use `AppSec::Configuration::Settings`

Reply:
I guess `Datadog::AppSec::Configuration::Settings.new.merge(Datadog::AppSec::Configuration::DSL.new.tap { |it| it.enabled = true })` works but it does have a bit of a code smell :)

Reply:
I simplify it by doing the following: 

```ruby
it 'returns a Datadog::AppSec::Component instance' do
  Datadog.configuration.appsec.enabled = true
  component = described_class.build_appsec_component(Datadog.configuration)
  expect(component).to be_a(described_class)
end
```